manager/includes/document.parser.class.inc.php

```diff
@@ -456,15 +458,27 @@ public function getSettings()
         }
 
         // store base_url and base_path inside config array
-        $this->config['base_url'] = MODX_BASE_URL;
-        $this->config['base_path'] = MODX_BASE_PATH;
-        $this->config['site_url'] = MODX_SITE_URL;
-        $this->config['valid_hostnames'] = MODX_SITE_HOSTNAMES;
-        $this->config['site_manager_url'] = MODX_MANAGER_URL;
+        $this->config['base_url']          = MODX_BASE_URL;
+        $this->config['base_path']         = MODX_BASE_PATH;
+        $this->config['site_url']          = MODX_SITE_URL;
+        $this->config['valid_hostnames']   = MODX_SITE_HOSTNAMES;
+        $this->config['site_manager_url']  = MODX_MANAGER_URL;
         $this->config['site_manager_path'] = MODX_MANAGER_PATH;
-        $this->error_reporting = isset($this->config['error_reporting']) ? $this->config['error_reporting'] : 0;
-        $this->config['filemanager_path'] = str_replace('[(base_path)]', MODX_BASE_PATH, (isset($this->config['filemanager_path']) ? $this->config['filemanager_path'] : ''));
-        $this->config['rb_base_dir'] = str_replace('[(base_path)]', MODX_BASE_PATH, (isset($this->config['rb_base_dir']) ? $this->config['rb_base_dir'] : ''));
+        $this->error_reporting = $this->config['error_reporting'] ?? 0;
+        if(strpos($this->config['filemanager_path']??'', '[(base_path)]')!==false) {
+            $this->config['filemanager_path'] = str_replace(
+                '[(base_path)]',
+                MODX_BASE_PATH,
+                $this->config['filemanager_path']
+            );
+        }
+        if(strpos($this->config['rb_base_dir']??'','[(base_path)]')!==false) {
+            $this->config['rb_base_dir'] = str_replace(
+                '[(base_path)]',
+                MODX_BASE_PATH,
+                $this->config['rb_base_dir']
+            );
+        }
```

Comment:
Possible warnings about undefined $this->config['filemanager_path'] and $this->config['rb_base_dir']

Reply:
@Pathologic 

> if(strpos($this->config['filemanager_path']??'', '[(base_path)]')!==false) {

This line doubles as a define check, need more? I will fix it further if necessary.
Please give me instructions.

Reply:
https://github.com/evocms-community/evolution/pull/222#discussion_r1071457826 - it was reviewed by @64j 

$this->config['rb_base_dir'] = str_replace(
    '[(base_path)]',
    MODX_BASE_PATH,
    $this->config['rb_base_dir'] - this summons warning
);

Maybe it can be changed to:
```
$this->config['rb_base_dir'] = $this->config['rb_base_dir'] ?? '';
if(strpos($this->config['rb_base_dir'],'[(base_path)]')!==false) {
          $this->config['rb_base_dir'] = str_replace(
              '[(base_path)]',
              MODX_BASE_PATH,
              $this->config['rb_base_dir']
          );
      }

```

Reply:
@Pathologic 

@64j 's point is correct. But this time things are different. Seek advice from @64j.

Reply:
@Pathologic 
After thinking it over, your idea is good. It's a good thing that this method doubles as normalizing the values.